### ISSUE TITLE
Branescript attribute list definition

### ISSUE THREAD

**Issue by DanielVoogsgerd** [epi-project/brane-documentation#31]
The Branescript compiler [assumes](https://github.com/epi-project/brane/blob/35166c3fd343a86bed9d60ca4d011bf70ae128dc/brane-dsl/src/parser/bscript.rs#L238) that attribute lists are comma-separated. However, this is incompatible with the formal definition of the language:
```
      AttrList ::= IDENT LPAREN Literals RPAREN
      Literals ::= Literals Literal
                 | Literal
```

**Comment by Lut99** [comment]
Good catch again! I'm genuinely impressed by how thorough you're looking at this! :D

Anyway yes this doesn't make sense. I tried to emulate Rust syntax here, so it's definitely with commas. Let me check the whole attribute tree to check if it makes sense with what I'm actually parsing.

**Comment by Lut99** [comment]
OK, right, it was a bit simpler Rust syntax. Only the following is allowed:
```
#[key = value]
```
OR
```
#[name(lit1, lit2, lit3, ...)]
```
Note the contents of the second variant is only _literals_, not any expressions. Makes it easier to parse the on-attributes though xD

**Comment by Lut99** [comment]
Change
```
Literals ::= Literals Literal
```
to
```
Literals ::= Literals COMMA Literal
```
and it's fixed.
